### ISSUE TITLE
ERROR:root:An error occurred: index 4 is out of bounds for dimension 1 with size 2. In the case of confusion matrix calculation, if the model does not predict any bounding box in an image, Does the code handles this case?

### ISSUE THREAD

**Issue by HassanBinHaroon** [kaanakan/object_detection_confusion_matrix#20]
In the case of confusion matrix calculation, if the model does not predict any bounding box in an image, Does the code handles this case?

The Confusion Matrix Computation Code:

"def process_batch(self, detections, labels):
"""
Return intersection-over-union (Jaccard index) of boxes.
Both sets of boxes are expected to be in (x1, y1, x2, y2) format.
Arguments:
detections (Array[N, 6]), x1, y1, x2, y2, conf, class
labels (Array[M, 5]), class, x1, y1, x2, y2
Returns:
None, updates confusion matrix accordingly
"""
detections = detections[detections[:, 4] > self.conf]
gt_classes = labels[:, 0].int()
detection_classes = detections[:, 5].int()
iou = general.box_iou(labels[:, 1:], detections[:, :4])

x = torch.where(iou > self.iou_thres)
if x[0].shape[0]:
    matches = torch.cat((torch.stack(x, 1), iou[x[0], x[1]][:, None]), 1).cpu().numpy()
    if x[0].shape[0] > 1:
        matches = matches[matches[:, 2].argsort()[::-1]]
        matches = matches[np.unique(matches[:, 1], return_index=True)[1]]
        matches = matches[matches[:, 2].argsort()[::-1]]
        matches = matches[np.unique(matches[:, 0], return_index=True)[1]]
else:
    matches = np.zeros((0, 3))

n = matches.shape[0] > 0
m0, m1, _ = matches.transpose().astype(np.int16)
for i, gc in enumerate(gt_classes):
    j = m0 == i
    if n and sum(j) == 1:
        self.matrix[gc, detection_classes[m1[j]]] += 1  # correct
    else:
        self.matrix[self.nc, gc] += 1  # background FP

if n:
    for i, dc in enumerate(detection_classes):
        if not any(m1 == i):
            self.matrix[dc, self.nc] += 1  # background FN"
